### ISSUE TITLE
feat: RMSNorm Instead of LayerNorm

### ISSUE THREAD

**Issue by andrewwgordon** [andrewwgordon/mini-gpt#3]
Simpler, faster normalization (used in Llama):

**Comment by andrewwgordon** [comment]
```python
class RMSNorm(nn.Module):
    def __init__(self, dim, eps=1e-6):
        super().__init__()
        self.weight = nn.Parameter(torch.ones(dim))
        self.eps = eps
        
    def forward(self, x):
        norm = x.pow(2).mean(-1, keepdim=True).sqrt()
        return x / (norm + self.eps) * self.weight
```
